packages/vscode-extension/src/webview/views/PreviewView.tsx

```diff
@@ -170,9 +176,18 @@ function PreviewView() {
   }, []);
 
   function toggleRecording() {
+    if (!hasAccessToProFeatures(licenseStatus)) {
+      openPaywall();
+      return;
+    }
```

Comment:
This feels like a common-enough pattern to lift it into a higher-order function. Something like
```
function withPaywallGuard(fn: Function, licenseStatus) {
  return (...args) => {
    if (!hasAccessToProFeatures(licenseStatus)) {
      openPaywall();
      return;
    }
    return fn(...args);
  }
}
```
Or even in hook form:
```
function usePaywalledCallback(fn: Function, dependencies: unknown[]) {
  const store$ = useStore();
  const licenseStatus = use$(store$.license.status);
  return useCallback(withPaywallGuard(fn, license), [license, ...dependencies])
}
```